satpy/readers/seviri_l1b_icare.py

```diff
@@ -254,13 +254,13 @@ def get_dataset(self, ds_id, ds_info):
         offset = data.attrs.get('add_offset')
         scale_factor = data.attrs.get('scale_factor')
         data = data.where(data != fill)
-        data.values = data.values.astype(np.float32)
+        data.data = data.data.astype(np.float32)
         if scale_factor is not None and offset is not None:
-            data.values *= scale_factor
-            data.values += offset
+            data *= scale_factor
+            data += offset
             # Now we correct range from 0-1 to 0-100 for VIS:
             if ds_id['name'] in self.ref_bands:
-                data.values *= 100.
+                data *= 100.
```

Comment:
These in-place operations should be `data = data * scale_factor`, see https://docs.dask.org/en/stable/delayed-best-practices.html#don-t-mutate-inputs

Reply:
You would be replacing the original dask graph that might be used somewhere else as-is. This could have unintended side-effects.

Reply:
Got it, thanks.